### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by seperalta22**
# Math Magicians: Events

## Summary

This branch implements the calculator logic to the calculator component.

Some features that I have implemented are:

- Import the calculate and operate files.
- Set the state of the calculator.
- Add the handleClick function to the calculator component.
- Refactor the code to be more readable.

### CODE REVIEW

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @SergioPeralta22 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑🛑
yPkfDFf

- [ ] Please consider removing the `Live demo` section from the readme file. It is only required when you have a project that is deployed. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,39 +1,40 @@
...
+        {buttons.map((button) => (
+          <button
+            type="button"
+            key={button.name}
+            className={`${button.className} ${button.color}`}
+            onClick={handleClick}
+          >
+            {button.name}
+          </button>
+        ))}
```
Comment (TadesseDev) [inline src/components/Calculator.js]: - [ ] Dear @SergioPeralta22, You did great on the project, But one of the main goals for this project is to pass props between parent and child components. As part of that, we need to pass `props` to a child compone of a `Calculate.js` component. Please make sure `Props` are passed correctly to any child components of the `Calculate.js` component.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @SergioPeralta22 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (SergioPeralta22) [reply src/components/Calculator.js:37]: @TadesseDev 

Why are you trying to block this project, could you please indicate which requirement I have failed?

image

**Review by DuaneDave — APPROVED** [review]
Hi @SergioPeralta22 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Communication between parent and child via props is done correctly

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (SergioPeralta22) [reply src/components/Calculator.js]: Why did you take back my code review after harassing me via Slack?

Comment (seperalta22) [reply src/components/Calculator.js]: Why did you take again the revision of my code after harassing me via Slack?

**Pull request merged**